src/popup/bookmarkModule.js

```diff
@@ -302,18 +302,32 @@ document.addEventListener('DOMContentLoaded', () => {
     removeBookmarkImages();
   });
 
-  elements.deleteAllBookmarksButton.addEventListener('click', () => {
+  elements.deleteBookmarksButton.addEventListener('click', () => {
     chrome.storage.sync.get({ bookmarks: [] }, (items) => {
       const bookmarksArray = items.bookmarks;
-      if (bookmarksArray.length === 0) {
-        showNotification('No Bookmarks Available', 'negative', 'snackbar-bookmarks');
+      const bookmarkDOMArray = Object.values(bookmarkDOM);
+      bookmarkDOMArray.forEach((checkbox) => {
+        if (checkbox.checked) {
+          bookmarksArray[bookmarksArray.indexOf(checkbox.id)] = null;
+        }
+      });
+      const updatedBookmarksArray = [];
+      bookmarksArray.forEach((bookmark) => {
+        if (bookmark != null) {
+          updatedBookmarksArray.push(bookmark);
+        }
+      });
+      if (bookmarksArray.length === updatedBookmarksArray.length) {
+        showNotification('No bookmark selected', 'negative', 'snackbar-bookmarks');
       } else {
-        bookmarksArray.splice(0, bookmarksArray.length); // empty array
-        chrome.storage.sync.set({ bookmarks: bookmarksArray }, () => {
+        chrome.storage.sync.set({ bookmarks: updatedBookmarksArray }, () => {
           // restoring initial layout of bookmarks section
           removeBookmarkImages();
-          msnry.layout();
-          restoreInitialContent('bookmarks');
+          addSpinner(elements.spinnerPlaceholderBookmarks, 'original');
+          removeOldSearchResults();
+          removeLoaderAnimation();
+          restoreInitialContent('primary');
```

Comment:
The following function calls do not do anything important to achieve the current behavior. They are redundant.
- `removeOldSearchResults()`
- `removeLoaderAnimation()`
-  `restoreInitialContent('primary')`

But I would argue for a better solution.

**Current workflow**
Right now, even if the user deletes a single image, the whole bookmarks section is loaded again. Image having a very large number of images in this section. Reloading them after deleting a few images (relatively speaking) would lead to triggering 99% of the same API requests and consecutively loading the image thumbnails. Add to that a slow internet connection or a low-end computer. 

**Improvement**
We can take the help of the [masonry grid](https://masonry.desandro.com/) used by the extension. It has the capability to reorganize the images in the grid after some of them are deleted without loading the images again ([check this](https://codepen.io/desandro/pen/IBJeq)). This is exactly what we want. In fact, this property is used to remove the bookmarks when the user clicks on the bookmark icon on the images in the bookmarks section ([implemented here](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/popup/bookmarkModule.js#L45))

In short: Instead of reloading the images again after deleting, remove the selected images from the DOM and let the masonry grid does the heavy lifting of reorganizing the rest of the images.